### PR TITLE
Add package zopfli version 1.0.3

### PR DESCRIPTION
**Pull request by suresh-thelkar**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Add package zopfli version 1.0.3

###### Change Log  

- NA

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

Buddy Build
- [AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=294118&view=results)
- [ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=294119&view=results)

### CODE REVIEW

**Comment by suresh-thelkar** [comment]
Please note that I have not fixed the linter false warnings.

File: SPECS-EXTENDED/zopfli/zopfli.spec [hunk 1]
```diff
@@ -0,0 +1,134 @@
+%global __cmake_in_source_build 1
+%global so_ver 1.0.3
```
Comment (neha170) [inline SPECS-EXTENDED/zopfli/zopfli.spec:2]: nit: we can get rid of so_ver and use %version instead

**Review by neha170 — APPROVED** [review]
Looks good, but for a small comment

File: SPECS-EXTENDED/zopfli/zopfli.spec (hunk 1 above)
Comment (suresh-thelkar) [reply SPECS-EXTENDED/zopfli/zopfli.spec]: Thank you for the review and approval. I have implemented your review comment in the recent commits.  Ran buddy builds after that. Here are the results. Both of the builds passed.
- [AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=295357&view=results)
- [ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=295358&view=results)

**Pull request merged**